src/analysis/normed_space/multilinear.lean

```diff
@@ -0,0 +1,796 @@
+  bound.-/
+  have : continuous_at f 0 := hf.continuous_at,
+  rcases metric.tendsto_nhds_nhds.1 this 1 zero_lt_one with ⟨ε, ε_pos, hε⟩,
+  let δ := ε/2,
+  have δ_pos : δ > 0 := half_pos ε_pos,
+  /- On points of size at most `δ`, `f` is bounded (by `1 + ∥f 0∥`). -/
+  have H : ∀{a}, ∥a∥ ≤ δ → ∥f a∥ ≤ 1 + ∥f 0∥,
+  { assume a ha,
+    have : dist (f a) (f 0) ≤ 1,
+    { apply le_of_lt (hε _),
+      rw [dist_eq_norm, sub_zero],
+      exact lt_of_le_of_lt ha (half_lt_self ε_pos) },
+    calc ∥f a∥ = dist (f a) 0 : (dist_zero_right _).symm
+      ... ≤ dist (f a) (f 0) + dist (f 0) 0 : dist_triangle _ _ _
+      ... ≤ 1 + ∥f 0∥ : by { rw dist_zero_right, exact add_le_add_right this _ } },
+  rcases normed_field.exists_one_lt_norm 𝕜 with ⟨c, hc⟩,
+  set C := (1 + ∥f 0∥) * univ.prod (λ(i : ι), δ⁻¹ * ∥c∥),
+  have C_pos : 0 < C :=
+    mul_pos (lt_of_lt_of_le zero_lt_one (by simp))
+      (prod_pos (λi hi, mul_pos (inv_pos δ_pos) (lt_of_le_of_lt zero_le_one hc))),
+  refine ⟨C, C_pos, λm, _⟩,
+  /- Given a general point `m`, rescale each coordinate to bring it to `[δ/∥c∥, δ]` by multiplication
+  by a power of a scalar `c` with norm `∥c∥ > 1`.-/
+  by_cases h : ∃i, m i = 0,
+  { rcases h with ⟨i, hi⟩,
+    rw [f.map_coord_zero i hi, _root_.norm_zero],
+    exact mul_nonneg'  (le_of_lt C_pos) (prod_nonneg (λi hi, norm_nonneg _)) },
+  { push_neg at h,
+    have : ∀i, ∃d:𝕜, d ≠ 0 ∧ ∥d • (m i)∥ ≤ δ ∧ (δ/∥c∥ ≤ ∥d • m i∥) ∧ (∥d∥⁻¹ ≤ δ⁻¹ * ∥c∥ * ∥m i∥) :=
+      λi, rescale_to_shell hc δ_pos (h i),
+    choose d hd using this,
+    have A : 0 ≤ 1 + ∥f 0∥ := add_nonneg zero_le_one (norm_nonneg _),
+    have B : ∀ (i : ι), i ∈ univ → 0 ≤ ∥d i∥⁻¹ := λi hi, by simp,
+    -- use the bound on `f` on the ball of size `δ` to conclude.
+    calc
+      ∥f m∥ = ∥f (λi, (d i)⁻¹ • (d i • m i))∥ :
+        by { unfold_coes, congr, ext i, rw [← mul_smul, inv_mul_cancel (hd i).1, one_smul] }
+      ... = ∥univ.prod (λi, (d i)⁻¹) • f (λi, d i • m i)∥ : by rw f.map_smul_univ
+      ... = univ.prod (λi, ∥d i∥⁻¹) * ∥f (λi, d i • m i)∥ :
+        by { rw [norm_smul, normed_field.norm_prod], congr, ext i, rw normed_field.norm_inv }
+      ... ≤ univ.prod (λi, ∥d i∥⁻¹) * (1 + ∥f 0∥) :
+        mul_le_mul_of_nonneg_left (H ((pi_norm_le_iff (le_of_lt δ_pos)).2 (λi, (hd i).2.1)))
+          (prod_nonneg B)
+      ... ≤ univ.prod (λi, δ⁻¹ * ∥c∥ * ∥m i∥) * (1 + ∥f 0∥) :
+        mul_le_mul_of_nonneg_right (prod_le_prod B (λi hi, (hd i).2.2.2)) A
+      ... = univ.prod (λ(i : ι), δ⁻¹ * ∥c∥) * univ.prod (λi, ∥m i∥) * (1 + ∥f 0∥) :
+        by rw prod_mul_distrib
+      ... = C * univ.prod (λ (i : ι), ∥m i∥) :
+        by rw [mul_comm, ← mul_assoc] }
+end
+
+/-- If `f` satisfies a boundedness property around `0`, one can deduce a bound on `f m₁ - f m₂`
+using the multilinearity. Here, we give a precise but hard to use version. See
+`norm_image_sub_le_of_bound` for a less precise but more usable version. -/
```

Comment:
I think it will be easier to understand this formula if you formulate its specific case (n=2? n=3?) in the docstring.

Reply:
Done